src/components/Channel/Channel.js

```diff
@@ -1,148 +1,674 @@
-  };
-
-  static defaultProps = {
-    disableKeyboardCompatibleView: false,
-    disableIfFrozenChannel: true,
-    KeyboardCompatibleView,
-  };
-
-  render() {
-    const { t } = this.props;
-    if (!this.props.channel.cid) {
-      return <Text>{t('Please select a channel first')}</Text>;
+const Channel = (props) => {
+  const {
+    channel,
+    children,
+    disableIfFrozenChannel = true,
+    disableKeyboardCompatibleView = false,
+    emojiData = emojiDataDefault,
+    KeyboardCompatibleView = KeyboardCompatibleViewDefault,
+  } = props;
+
+  const { client, isOnline, logger = () => {} } = useContext(ChatContext);
+  const { t } = useContext(TranslationContext);
+
+  const [editing, setEditing] = useState(null);
+  const [error, setError] = useState(false);
+  const [eventHistory, setEventHistory] = useState({});
+  /**
+   * We save the events in state so that we can display event message
+   * next to the message after which it was received, in MessageList.
+   *
+   * e.g., eventHistory = {
+   *   message_id_1: [
+   *     { ...event_obj_received_after_message_id_1__1 },
+   *     { ...event_obj_received_after_message_id_1__2 },
+   *     { ...event_obj_received_after_message_id_1__3 },
+   *   ],
+   *   message_id_2: [
+   *     { ...event_obj_received_after_message_id_2__1 },
+   *     { ...event_obj_received_after_message_id_2__2 },
+   *     { ...event_obj_received_after_message_id_2__3 },
+   *   ]
+   * }
+   */
+  const [hasMore, setHasMore] = useState(true);
+  const [lastRead, setLastRead] = useState();
+  const [loading, setLoading] = useState(true);
+  const [loadingMore, setLoadingMore] = useState(false);
+  const [members, setMembers] = useState(Immutable({}));
+  const [messages, setMessages] = useState(Immutable([]));
+  const [online, setOnline] = useState(isOnline);
+  const [read, setRead] = useState(Immutable({}));
+  const [thread, setThread] = useState(props.thread);
+  const [threadHasMore, setThreadHasMore] = useState(true);
+  const [threadLoadingMore, setThreadLoadingMore] = useState(false);
+  const [threadMessages, setThreadMessages] = useState(
+    channel.state?.threads?.[props.thread?.id] || [],
+  );
+  const [typing, setTyping] = useState(Immutable({}));
+  const [unmounted, setUnmounted] = useState(false);
+  const [watcherCount, setWatcherCount] = useState();
+  const [watchers, setWatchers] = useState(Immutable({}));
+
+  useEffect(() => {
+    logger('Channel component', 'component mount', {
+      tags: ['lifecycle', 'channel'],
+      props,
+    });
+
+    if (channel) initChannel();
+
+    return () => {
+      logger('Channel component', 'component unmount', {
+        tags: ['lifecycle', 'channel'],
+        props,
+      });
+
+      client.off('connection.recovered', handleEvent);
+      channel.off?.(handleEvent);
+      handleEventStateThrottled.cancel();
+      loadMoreFinishedDebounced.cancel();
+      loadMoreThreadFinishedDebounced.cancel();
+      setUnmounted(true);
```

Comment:
what is tracking the mounted state helping us with in a functional component? shouldn't we be able to avoid all executions in a functional component if the component is unmounted without tracking it in a hook from within in the useEffect cleanup?

Reply:
tbh not totally sure, just carrying this logic forward. figured there was a reason we were tracking unmounted state so kept it in there

Reply:
Don't think we need it anymore. Lets remove `unmounted` logic.